File: source/clibCoreCs/Models/Entity/libraryFilesModel.cs
//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

using System;
using System.Collections.Generic;
using System.Text;
using Contensive.BaseClasses;
using Contensive.Core.Controllers;
using Newtonsoft.Json;

namespace Contensive.Core.Models.Entity
{
	public class libraryFilesModel : baseModel
	{
		//
		//====================================================================================================
		//-- const
		public const string contentName = "library Files";
		public const string contentTableName = "cclibraryFiles";
		private const string contentDataSource = "default";
		//
		//====================================================================================================
		// -- instance properties
		public string AltSizeList {get; set;}
		public string AltText {get; set;}
		public int Clicks {get; set;}
		public string Description {get; set;}
		public string Filename {get; set;}
		public int FileSize {get; set;}
		public int FileTypeID {get; set;}
		public int FolderID {get; set;}
		public int pxHeight {get; set;}
		public int pxWidth {get; set;}
		//
		//====================================================================================================
		public static libraryFilesModel add(coreClass cpCore)
		{
			return add<libraryFilesModel>(cpCore);
		}
		//
		//====================================================================================================
		public static libraryFilesModel add(coreClass cpCore, ref List<string> callersCacheNameList)
		{
			return add<libraryFilesModel>(cpCore, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static libraryFilesModel create(coreClass cpCore, int recordId)
		{
			return create<libraryFilesModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static libraryFilesModel create(coreClass cpCore, int recordId, ref List<string> callersCacheNameList)
		{
			return create<libraryFilesModel>(cpCore, recordId, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static libraryFilesModel create(coreClass cpCore, string recordGuid)
		{
			return create<libraryFilesModel>(cpCore, recordGuid);
		}
		//
		//====================================================================================================
		public static libraryFilesModel create(coreClass cpCore, string recordGuid, ref List<string> callersCacheNameList)
		{
			return create<libraryFilesModel>(cpCore, recordGuid, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static libraryFilesModel createByName(coreClass cpCore, string recordName)
		{
			return createByName<libraryFilesModel>(cpCore, recordName);
		}
		//
		//====================================================================================================
		public static libraryFilesModel createByName(coreClass cpCore, string recordName, ref List<string> callersCacheNameList)
		{
			return createByName<libraryFilesModel>(cpCore, recordName, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public void save(coreClass cpCore)
		{
			base.save(cpCore);
		}
		//
		//====================================================================================================
		public static void delete(coreClass cpCore, int recordId)
		{
			delete<libraryFilesModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static void delete(coreClass cpCore, string ccGuid)
		{
			delete<libraryFilesModel>(cpCore, ccGuid);
		}
		//
		//====================================================================================================
		public static List<libraryFilesModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy, List<string> callersCacheNameList)
		{
			return createList<libraryFilesModel>(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList);
		}
		//
		//====================================================================================================
		public static List<libraryFilesModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy)
		{
			return createList<libraryFilesModel>(cpCore, sqlCriteria, sqlOrderBy);
		}
		//
		//====================================================================================================
		public static List<libraryFilesModel> createList(coreClass cpCore, string sqlCriteria)
		{
			return createList<libraryFilesModel>(cpCore, sqlCriteria);
		}
		//
		//====================================================================================================
		public void invalidatePrimaryCache(coreClass cpCore, int recordId)
		{
			invalidateCacheSingleRecord<libraryFilesModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static string getRecordName(coreClass cpcore, int recordId)
		{
			return baseModel.getRecordName<libraryFilesModel>(cpcore, recordId);
		}
		//
		//====================================================================================================
		public static string getRecordName(coreClass cpcore, string ccGuid)
		{
			return baseModel.getRecordName<libraryFilesModel>(cpcore, ccGuid);
		}
		//
		//====================================================================================================
		public static int getRecordId(coreClass cpcore, string ccGuid)
		{
			return baseModel.getRecordId<libraryFilesModel>(cpcore, ccGuid);
		}
		//
		//====================================================================================================
		public static libraryFilesModel createDefault(coreClass cpcore)
		{
			return createDefault<libraryFilesModel>(cpcore);
		}
	}
}

File: source/processor/Models/Db/1/libraryFilesModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    Public Class libraryFilesModel
        Inherits baseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "library Files"
        Public Const contentTableName As String = "cclibraryFiles"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        Public Property AltSizeList As String
        Public Property AltText As String
        Public Property Clicks As Integer
        Public Property Description As String
        Public Property Filename As String
        Public Property FileSize As Integer
        Public Property FileTypeID As Integer
        Public Property FolderID As Integer
        Public Property pxHeight As Integer
        Public Property pxWidth As Integer
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass) As libraryFilesModel
            Return add(Of libraryFilesModel)(cpCore)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass, ByRef callersCacheNameList As List(Of String)) As libraryFilesModel
            Return add(Of libraryFilesModel)(cpCore, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer) As libraryFilesModel
            Return create(Of libraryFilesModel)(cpCore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer, ByRef callersCacheNameList As List(Of String)) As libraryFilesModel
            Return create(Of libraryFilesModel)(cpCore, recordId, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String) As libraryFilesModel
            Return create(Of libraryFilesModel)(cpCore, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String, ByRef callersCacheNameList As List(Of String)) As libraryFilesModel
            Return create(Of libraryFilesModel)(cpCore, recordGuid, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String) As libraryFilesModel
            Return createByName(Of libraryFilesModel)(cpCore, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String, ByRef callersCacheNameList As List(Of String)) As libraryFilesModel
            Return createByName(Of libraryFilesModel)(cpCore, recordName, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cpCore As coreClass)
            MyBase.save(cpCore)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, recordId As Integer)
            delete(Of libraryFilesModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, ccGuid As String)
            delete(Of libraryFilesModel)(cpCore, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String, callersCacheNameList As List(Of String)) As List(Of libraryFilesModel)
            Return createList(Of libraryFilesModel)(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String) As List(Of libraryFilesModel)
            Return createList(Of libraryFilesModel)(cpCore, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String) As List(Of libraryFilesModel)
            Return createList(Of libraryFilesModel)(cpCore, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub invalidatePrimaryCache(cpCore As coreClass, recordId As Integer)
            invalidateCacheSingleRecord(Of libraryFilesModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, recordId As Integer) As String
            Return baseModel.getRecordName(Of libraryFilesModel)(cpcore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, ccGuid As String) As String
            Return baseModel.getRecordName(Of libraryFilesModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cpcore As coreClass, ccGuid As String) As Integer
            Return baseModel.getRecordId(Of libraryFilesModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createDefault(cpcore As coreClass) As libraryFilesModel
            Return createDefault(Of libraryFilesModel)(cpcore)
        End Function
    End Class
End Namespace
